### ISSUE TITLE
[CT-1819] [Bug] integration_results.csv and unit_results.csv have missing data

### ISSUE THREAD

**Issue by MichelleArk** [dbt-labs/dbt-core#6611]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Currently, GHA jobs for integration and unit test steps create .csv file artifacts collected during unit and integration test runs (unit_results.csv, integration_results.csv). Both these files only contain the pytest duration data associated with the last run pytest command. 

For unit tests, this means that `test/unit` data is missing from unit_results.csv. For integration tests, `test/integration` and `test/functional` data is missing from integration_results.csv. 

This is occurring because the same [csv file name is being passed](https://github.com/dbt-labs/dbt-core/blob/main/.github/workflows/main.yml#L131) to [each tox command](https://github.com/dbt-labs/dbt-core/blob/main/tox.ini#L28-L30), overwriting any previously written data to the file. 

### Expected Behavior

No missing test duration data for any `pytest` command run in CI. Whether the final artifact is a single unified file, or through a collection of module-specific files is unimportant given these artifacts have no downstream dependencies. 

### Steps To Reproduce

In a local environment, run `TOXENV='integration' PYTEST_ADDOPTS="--csv integration_results.csv" tox`. Resulting `integration_results.csv` file only contains `tests/adapter` data. 

In CI: navigate to artifacts and download any `integration_results_*.csv` file from a CI run. E.g. https://github.com/dbt-labs/dbt-core/actions/runs/3912846819

**Comment by nathaniel-may** [comment]
Possible solutions:
- fix the file overwriting
- ship results to datadog instead of relying on the file
- use dbt???

**Comment by iknox-fa** [comment]
Per BLG 1/23/23 Blocked on a decision around the tool we'll use to consume this data.  @peterallenwebb can you update this ticket when that decision is made please?

**Comment by MichelleArk** [comment]
No longer relevant as we are not uploading the integration/unit_results.csv files after switching to publishing coverage data to codecov: https://github.com/dbt-labs/dbt-core/pull/8127/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3L188
